app/components/views/TicketsPage/index.js

```diff
@@ -1,183 +1,78 @@
-import React, { Component } from "react";
+// @flow
+import Header from "./Header";
+import { RouteTransition } from "shared";
+
+const enterLeft = { atEnter: { offset: -100 }, atActive: { offset: 0 }, atLeave: { offset: 100 }, mapStyles: (styles) => ({ transform: `translateX(${styles.offset}%)` })};
+const enterRight = { atEnter: { offset: 100 }, atActive: { offset: 0 }, atLeave: { offset: -100 }, mapStyles: (styles) => ({ transform: `translateX(${styles.offset}%)` })};
+const opts = { stiffness: 150, damping: 20 };
+
+const wrapperComponent = props => <div className="tab-content" { ...props } />;
+
+const Tickets = ({ children, location }) => {
+  const tabs = ["purchase", "mytickets", "governance", "statistics"];
+  const page = "tickets";
+  // this will be removed w/ react router 4
+  const pathname = location.pathname.split("/")[2];
+  const effect = pathname === "purchase" ? enterLeft : enterRight;
```

Comment:
if the pathname is tabs[0], enterLeft.
if the pathname is tabs[tabs.length -1], enter right.

if the pathname is anything in between those, use location.history to check what the previous one was (using the same .split("/")[2] ).
if tabs.indexOf(prev) >tabs.indexOf(pathname), enterLeft, else enter right.

Reply:
cool thx.  was missing the location.history piece for this.

Reply:
add a breakpoint to the code in the browser right there so you can play around w/ the location.history object to figure out how to extract the previous pathname from it. I can't recall off the top of my head.